### PR TITLE
run function update

### PR DESCRIPTION
**Pull request by gerbsec**
With this we get more out of run, and it doesn't limit us to cmd.exe or bash. Cross platform as well.

### CODE REVIEW

File: endpoint/endpoint.go [hunk 1]
```diff
@@ -135,23 +134,15 @@ func Serve(address string, protocol string) {
...
-		cmd := exec.Command("bash", "-c", command)
-		stdout, err := cmd.Output()
-		if err != nil {
-			println(err.Error())
-			os.Exit(1)
+func Run(command string, args []string) (int, string, string) {
+	cmd := exec.Command(command, args...)
+	stdout, err := cmd.Output()
+	if err != nil {
+		if exitError, ok := err.(*exec.ExitError); ok {
```
Comment (privateducky) [inline endpoint/endpoint.go:141]: nice change. 
@bfuzzy1 @gerbsec can you give a confirmation this works with our existing tests, if applied?
if so, we should group those test changes into this PR. the code here is good.

Comment (bfuzzy1) [reply endpoint/endpoint.go:141]: Manual testing complete. Probe testing next week.

File: endpoint/endpoint.go [hunk 2]
```diff
@@ -135,23 +134,15 @@ func Serve(address string, protocol string) {
...
+			return exitError.ExitCode(), string(stdout), string(exitError.Stderr)
+		} else {
+			return 1, string(stdout), err.Error()
```
Comment (privateducky) [inline endpoint/endpoint.go:144]: what's a case you land inside this else block?

File: tests/021a1dbf-e080-495c-937e-7fa79d39bf30/021a1dbf-e080-495c-937e-7fa79d39bf30.go [hunk 3]
```diff
@@ -6,36 +6,52 @@ CREATED: 2023-01-21
...
 
 import (
-	"github.com/preludeorg/test/endpoint"
 	"runtime"
+
+	Endpoint "github.com/preludeorg/test/endpoint"
 )
 
-func command1() string {
+func executor() string {
```
Comment (privateducky) [inline tests/021a1dbf-e080-495c-937e-7fa79d39bf30/021a1dbf-e080-495c-937e-7fa79d39bf30.go:14]: this function would be nicely served in the Endpoint module as: 'Endpoint.Shell()'

File: tests/021a1dbf-e080-495c-937e-7fa79d39bf30/021a1dbf-e080-495c-937e-7fa79d39bf30.go [hunk 4]
```diff
@@ -6,36 +6,52 @@ CREATED: 2023-01-21
...
+	println(stdout)
+
+	exitCode2, stdout2, stderr2 := Endpoint.Run(executor(), command2())
+	if exitCode2 != 0 {
+		println(stderr2)
+		Endpoint.Stop(1)
 	}
-	Endpoint.Stop(101)
+	println(stdout2)
+	Endpoint.Stop(100)
```
Comment (privateducky) [inline tests/021a1dbf-e080-495c-937e-7fa79d39bf30/021a1dbf-e080-495c-937e-7fa79d39bf30.go:54]: Code 100 means this test completed and it was good for the customer.
101 means the test completed but something should have stopped it.
why change it to 100?

Comment (privateducky) [reply tests/021a1dbf-e080-495c-937e-7fa79d39bf30/021a1dbf-e080-495c-937e-7fa79d39bf30.go:54]: same to other changed tests

File: tests/021a1dbf-e080-495c-937e-7fa79d39bf30/021a1dbf-e080-495c-937e-7fa79d39bf30.go [hunk 5]
```diff
@@ -6,36 +6,52 @@ CREATED: 2023-01-21
...
 	}
 }
 
 func test() {
-	response1 := Endpoint.Run(command1())
-	print(response1)
-	if len(response1) > 0 {
-		response2 := Endpoint.Run(command2())
-		print(response2)
+	exitCode, stdout, stderr := Endpoint.Run(executor(), command1())
```
Comment (privateducky) [inline tests/021a1dbf-e080-495c-937e-7fa79d39bf30/021a1dbf-e080-495c-937e-7fa79d39bf30.go:41]: I'm worried that sending both of these params to the Run function over time will become tiring. I'll help think of an alternative.

**Review by privateducky — CHANGES_REQUESTED** [review]
Some modifications mentioned. 
Also: please run a linter on this. 
Finally-- does this solve any immediate problems with our existing tests or just prepare ourselves for later ones?

File: tests/53ff0bda-537e-4493-90cf-220145d9569a/53ff0bda-537e-4493-90cf-220145d9569a.go [hunk 6]
```diff
@@ -6,24 +6,38 @@ CREATED: 2023-01-06 10:54:04.264000
...
+
+	Endpoint "github.com/preludeorg/test/endpoint"
 )
 
 func test() {
 	if runtime.GOOS == "windows" {
-		encoded := "dwBoAG8AYQBtAGkA"
-		task := fmt.Sprintf("powershell.exe -e %s", encoded)
-		stdout := Endpoint.Run(task)
+		encoded := "ZQBjAGgAbwAgAC0AbgAgAFAAcgBlAGwAdQBkAGUA"
```
Comment (privateducky) [inline tests/53ff0bda-537e-4493-90cf-220145d9569a/53ff0bda-537e-4493-90cf-220145d9569a.go:17]: this test is changing too many things.. the run function, exit codes, the encoded string in this case... focus this PR on 1 change

**Review by privateducky — COMMENTED** [review]
the formatting on this file seems way off or is that just me?

File: tests/53ff0bda-537e-4493-90cf-220145d9569a/53ff0bda-537e-4493-90cf-220145d9569a.go [hunk 7]
```diff
@@ -6,24 +6,38 @@ CREATED: 2023-01-06 10:54:04.264000
 package main
 
 import (
-	"fmt"
-	"github.com/preludeorg/test/endpoint"
 	"runtime"
+	"strings"
+
+	Endpoint "github.com/preludeorg/test/endpoint"
```
Comment (privateducky) [inline tests/53ff0bda-537e-4493-90cf-220145d9569a/53ff0bda-537e-4493-90cf-220145d9569a.go:12]: why include this import change? it seems disconnected from the primary change set here.

**Comment by privateducky** [comment]
closing in favor of: https://github.com/preludeorg/test/pull/51

note there are some changes in here that are not true to the title of this PR. @gerbsec you should cherry pick out anything that is important.

**Pull request closed without merging**